From c9900113e5753e97550262a5474f9644ff258d18 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 12 Apr 2016 11:08:33 +1000
Subject: [PATCH 0551/1722] drm/amd/dal: move some defines and removed unused
 struct

This removes some unused things from the adapter service.

Signed-off-by: Dave Airlie <airlied@redhat.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 .../gpu/drm/amd/dal/dc/adapter/adapter_service.c   |   8 +
 .../gpu/drm/amd/dal/dc/adapter/adapter_service.h   |   2 +
 drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c      | 193 ---------------------
 .../amd/dal/include/adapter_service_interface.h    |  13 --
 .../drm/amd/dal/include/grph_object_ctrl_defs.h    |   5 -
 5 files changed, 10 insertions(+), 211 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
index 3a26c84..a7b6345 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
@@ -55,6 +55,14 @@
 
 #include "diagnostics/hw_ctx_adapter_service_diag.h"
 
+#define SIZEOF_BACKLIGHT_LUT 101
+#define ABSOLUTE_BACKLIGHT_MAX 255
+#define DEFAULT_MIN_BACKLIGHT 12
+#define DEFAULT_MAX_BACKLIGHT 255
+#define BACKLIGHT_CURVE_COEFFB 100
+#define BACKLIGHT_CURVE_COEFFA_FACTOR 10000
+#define BACKLIGHT_CURVE_COEFFB_FACTOR 100
+
 /*
  * Adapter service feature entry table.
  *
diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.h b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.h
index 57f9a87..631e28d 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.h
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.h
@@ -31,6 +31,8 @@
 #include "include/adapter_service_interface.h"
 #include "wireless_data_source.h"
 
+#define SIZEOF_BACKLIGHT_LUT 101
+
 /*
  * Forward declaration
  */
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
index 9c8296b..9096b51 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
@@ -58,8 +58,6 @@ static const uint8_t ext_display_connection_guid[NUMBER_OF_UCHAR_FOR_GUID] = {
 
 static uint8_t *get_image(struct bios_parser *bp, uint32_t offset,
 	uint32_t size);
-static uint32_t get_record_size(uint8_t *record);
-static uint32_t get_edid_size(const ATOM_FAKE_EDID_PATCH_RECORD *edid);
 static enum object_type object_type_from_bios_object_id(
 	uint32_t bios_object_id);
 static struct graphics_object_id object_id_from_bios_object_id(
@@ -2442,87 +2440,6 @@ static uint32_t get_ss_entry_number_from_internal_ss_info_tbl_V3_1(
 	return number;
 }
 
-static ATOM_FAKE_EDID_PATCH_RECORD *get_faked_edid_record(
-	struct bios_parser *bp)
-{
-	uint32_t size;
-	uint8_t *record;
-	ATOM_LVDS_INFO_V12 *info;
-
-	if (!DATA_TABLES(LVDS_Info))
-		return NULL;
-
-	info = GET_IMAGE(ATOM_LVDS_INFO_V12, DATA_TABLES(LVDS_Info));
-
-	if (!info)
-		return NULL;
-
-	if (1 != info->sHeader.ucTableFormatRevision
-			|| 2 > info->sHeader.ucTableContentRevision)
-		return NULL;
-
-	if (!le16_to_cpu(info->usExtInfoTableOffset))
-		return NULL;
-
-	record = GET_IMAGE(uint8_t, DATA_TABLES(LVDS_Info)
-			+ le16_to_cpu(info->usExtInfoTableOffset));
-
-	if (!record)
-		return NULL;
-
-	for (;;) {
-		if (ATOM_RECORD_END_TYPE == *record)
-			return NULL;
-
-		if (LCD_FAKE_EDID_PATCH_RECORD_TYPE == *record)
-			break;
-
-		size = get_record_size(record);
-
-		if (!size)
-			return NULL;
-
-		record += size;
-	}
-
-	return (ATOM_FAKE_EDID_PATCH_RECORD *)record;
-}
-
-enum bp_result dc_bios_get_faked_edid_len(struct dc_bios *dcb,
-					  uint32_t *len)
-{
-	struct bios_parser *bp = BP_FROM_DCB(dcb);
-	ATOM_FAKE_EDID_PATCH_RECORD *edid_record = get_faked_edid_record(bp);
-
-	if (!edid_record)
-		return BP_RESULT_NORECORD;
-
-	*len = get_edid_size(edid_record);
-
-	return BP_RESULT_OK;
-}
-
-enum bp_result dc_bios_get_faked_edid_buf(struct dc_bios *dcb,
-					  uint8_t *buff,
-					  uint32_t len)
-{
-	struct bios_parser *bp = BP_FROM_DCB(dcb);
-	ATOM_FAKE_EDID_PATCH_RECORD *edid_record = get_faked_edid_record(bp);
-	uint32_t edid_size;
-
-	if (!edid_record)
-		return BP_RESULT_NORECORD;
-
-	edid_size = get_edid_size(edid_record);
-
-	if (len < edid_size)
-		return BP_RESULT_BADINPUT; /* buffer not big enough to fill */
-
-	memmove(buff, &edid_record->ucFakeEDIDString, edid_size);
-
-	return BP_RESULT_OK;
-}
-
 /**
  * bios_parser_get_gpio_pin_info
  * Get GpioPin information of input gpio id
@@ -2585,77 +2502,6 @@ enum bp_result dc_bios_get_gpio_pin_info(
 	return BP_RESULT_NORECORD;
 }
 
-/**
- * BiosParserObject::EnumEmbeddedPanelPatchMode
- * Get embedded panel patch mode
- *
- * @param index, mode index
- * @param info, embedded panel patch mode structure
- * @return Bios parser result code
- */
-enum bp_result dc_bios_enum_embedded_panel_patch_mode(struct dc_bios *dcb,
-						      uint32_t index,
-						      struct embedded_panel_patch_mode *mode)
-{
-	struct bios_parser *bp = BP_FROM_DCB(dcb);
-	uint32_t record_size;
-	uint32_t record_index;
-	uint8_t *record;
-	ATOM_LVDS_INFO_V12 *info;
-	ATOM_PATCH_RECORD_MODE *mode_record;
-	ATOM_MASTER_LIST_OF_DATA_TABLES *list_of_tables;
-
-	if (!mode)
-		return BP_RESULT_BADINPUT;
-
-	list_of_tables = &bp->master_data_tbl->ListOfDataTables;
-	if (!list_of_tables->LVDS_Info)
-		return BP_RESULT_UNSUPPORTED;
-
-	info = GET_IMAGE(ATOM_LVDS_INFO_V12, list_of_tables->LVDS_Info);
-
-	if (!info)
-		return BP_RESULT_BADBIOSTABLE;
-
-	if (1 != info->sHeader.ucTableFormatRevision
-			|| 2 > info->sHeader.ucTableContentRevision)
-		return BP_RESULT_UNSUPPORTED;
-
-	if (!le16_to_cpu(info->usExtInfoTableOffset))
-		return BP_RESULT_UNSUPPORTED;
-
-	record = GET_IMAGE(uint8_t, list_of_tables->LVDS_Info +
-			le16_to_cpu(info->usExtInfoTableOffset));
-
-	if (!record)
-		return BP_RESULT_BADBIOSTABLE;
-
-	for (record_index = 0;;) {
-		if (ATOM_RECORD_END_TYPE == *record)
-			return BP_RESULT_NORECORD;
-
-		if (LCD_MODE_PATCH_RECORD_MODE_TYPE == *record) {
-			if (record_index == index)
-				break;
-			record_index++;
-		}
-
-		record_size = get_record_size(record);
-
-		if (!record_size)
-			return BP_RESULT_NORECORD;
-
-		record += record_size;
-	}
-
-	mode_record = (ATOM_PATCH_RECORD_MODE *) record;
-
-	mode->width = le16_to_cpu(mode_record->usHDisp);
-	mode->height = le16_to_cpu(mode_record->usVDisp);
-
-	return BP_RESULT_OK;
-}
-
 static enum bp_result get_gpio_i2c_info(struct bios_parser *bp,
 	ATOM_I2C_RECORD *record,
 	struct graphics_object_i2c_info *info)
@@ -2878,45 +2724,6 @@ static uint8_t *get_image(struct bios_parser *bp,
 		return NULL;
 }
 
-static uint32_t get_record_size(uint8_t *record)
-{
-	switch (*record) {
-	case LCD_MODE_PATCH_RECORD_MODE_TYPE:
-		return sizeof(ATOM_PATCH_RECORD_MODE);
-
-	case LCD_RTS_RECORD_TYPE:
-		return sizeof(ATOM_LCD_RTS_RECORD);
-
-	case LCD_CAP_RECORD_TYPE:
-		return sizeof(ATOM_LCD_MODE_CONTROL_CAP);
-
-	case LCD_FAKE_EDID_PATCH_RECORD_TYPE: {
-		ATOM_FAKE_EDID_PATCH_RECORD *fake_record =
-				(ATOM_FAKE_EDID_PATCH_RECORD *) record;
-		uint32_t edid_size = get_edid_size(fake_record);
-
-		return sizeof(ATOM_FAKE_EDID_PATCH_RECORD) + edid_size
-				- sizeof(fake_record->ucFakeEDIDString);
-	}
-
-	case LCD_PANEL_RESOLUTION_RECORD_TYPE:
-		return sizeof(ATOM_PANEL_RESOLUTION_PATCH_RECORD);
-
-	default:
-		return 0;
-	}
-}
-
-static uint32_t get_edid_size(const ATOM_FAKE_EDID_PATCH_RECORD *edid)
-{
-	uint32_t length = edid->ucFakeEDIDLength;
-
-	if (length < 128)
-		length = length * 128;
-
-	return length;
-}
-
 static struct graphics_object_id object_id_from_bios_object_id(
 	uint32_t bios_object_id)
 {
diff --git a/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h b/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
index 72dd7ad..94424dd 100644
--- a/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
+++ b/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
@@ -479,19 +479,6 @@ struct ddc *dal_adapter_service_obtain_ddc_from_i2c_info(
 	struct adapter_service *as,
 	struct graphics_object_i2c_info *info);
 
-#define SIZEOF_BACKLIGHT_LUT 101
-#define ABSOLUTE_BACKLIGHT_MAX 255
-#define DEFAULT_MIN_BACKLIGHT 12
-#define DEFAULT_MAX_BACKLIGHT 255
-#define BACKLIGHT_CURVE_COEFFB 100
-#define BACKLIGHT_CURVE_COEFFA_FACTOR 10000
-#define BACKLIGHT_CURVE_COEFFB_FACTOR 100
-
-struct panel_backlight_levels {
-	uint32_t ac_level_percentage;
-	uint32_t dc_level_percentage;
-};
-
 uint32_t dal_adapter_service_get_downscale_limit(
 		struct adapter_service *as);
 
diff --git a/drivers/gpu/drm/amd/dal/include/grph_object_ctrl_defs.h b/drivers/gpu/drm/amd/dal/include/grph_object_ctrl_defs.h
index 50e3f0a..6bb3d3a 100644
--- a/drivers/gpu/drm/amd/dal/include/grph_object_ctrl_defs.h
+++ b/drivers/gpu/drm/amd/dal/include/grph_object_ctrl_defs.h
@@ -163,11 +163,6 @@ struct embedded_panel_info {
 	uint32_t min_drr_refresh_rate;
 };
 
-struct embedded_panel_patch_mode {
-	uint32_t width;
-	uint32_t height;
-};
-
 struct firmware_info {
 	struct pll_info {
 		uint32_t crystal_frequency; /* in KHz */
-- 
2.7.4

